server/auth/store.go

```diff
@@ -493,11 +493,9 @@ func (as *authStore) UserChangePassword(r *pb.AuthUserChangePasswordRequest) (*p
 	var password []byte
 	var err error
 
-	if !user.Options.NoPassword {
```

Comment:
why not `user.Options != nil && !user.Options.NoPassword`?

Reply:
yes, at the beginning, I fixed it like this. After checking the code carefully, I found that this if check is unnecessary.

Reply:
> why not `user.Options != nil && !user.Options.NoPassword`?

In this way, changing the password will not succeed, and the final passwd is an empty password

Reply:
After second thought, I think the correct way should be `user.Options == nil || !user.Options.NoPassword`.  If `user.Options` is `nil`, then it means the etcd is at a version older than https://github.com/etcd-io/etcd/pull/9817; It means each user must have a password.

Reply:
good catch. however, in the UserChangePassword function, it seems that there will never be an empty password.  I chose the current optimization solution based on the simplicity of the code. can you give a question about the current solution?  If --no-passwd is specified when creating user, is it not allowed to change the password?

Reply:
> If --no-passwd is specified when creating user, is it not allowed to change the password?

In such case, the users should use CN based authentication. There should be no way to change password for such user. cc @mitake to double confirm.

Reply:
There is also a question. At present, there is no error to change the password for no password user, it seems succ.  In fact, it is failed. do we need to add an error code here?
```
% etcdctl user add test --no-password
User test created
% etcdctl user passwd test
Password of test:
Type password of test again for confirmation:
Password updated
```

Reply:
> do we need to add an error code here?

Agreed. Defer to @mitake

Reply:
Thanks a lot for fixing the issue @tangcong !

> If --no-passwd is specified when creating user, is it not allowed to change the password?

Yes, that should be the assumed design. Once a user is created, its `authpb.User.Option` doesn't have a way to be changed.

> do we need to add an error code here?

That's also true, the new password will be simply discarded. Showing an error is much better.

Reply:
ok, I will do it in another PR, this PR will solve the panic problem first, and we need to cherry-pick it to 3.5.  PR has updated and added a unit test case. without this PR, the new unit test cases will not pass.
```

--- FAIL: TestUserChangePassword (0.01s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x166f6c5]
```

Reply:
Understood, yeah it's great if you can open in another PR.

Reply:
@tangcong if you can open additional PR, that's really great. But feel free to assign it to me if you are busy.